File: android/MqttClient/app/src/main/java/com/example/mqttclient/MainActivity.java
package com.example.mqttclient;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import java.util.UUID;

public class MainActivity extends AppCompatActivity {
    MqttHelper mqttHelper;
    private final String serverUri = "soddong.duckdns.org:8883";
    private final String topic = "Starry/Pub";

    @SuppressLint("ResourceType")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mqttHelper = new MqttHelper(getApplicationContext(), serverUri, UUID.randomUUID().toString(), topic);

        EditText editText = findViewById(R.id.messageInput);

        Button sendBtn = findViewById(R.id.sendButton);
        sendBtn.setOnClickListener(v -> {
            String msg = String.valueOf(editText.getText());
            mqttHelper.publish(topic, msg);
        });
    }
}

File: android-app/app/src/main/java/com/soddong/myhome/MainActivity.kt
package com.soddong.myhome

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 에어컨 버튼 클릭 이벤트
        findViewById<MaterialButton>(R.id.airconButton).setOnClickListener {
            startActivity(Intent(this, AirconActivity::class.java))
        }

        // TV 버튼 클릭 이벤트
        findViewById<MaterialButton>(R.id.tvButton).setOnClickListener {
            // TODO: TV 액티비티 구현
        }

        // 조명 버튼 클릭 이벤트
        findViewById<MaterialButton>(R.id.lightButton).setOnClickListener {
            // TODO: 조명 액티비티 구현
        }

        // 스피커 버튼 클릭 이벤트
        findViewById<MaterialButton>(R.id.speakerButton).setOnClickListener {
            // TODO: 스피커 액티비티 구현
        }
    }
} 